app/controllers/warehouses_controller.rb

```diff
@@ -0,0 +1,74 @@
+class WarehousesController < ApplicationController
+  before_action :set_warehouse, only: [:show, :edit, :update, :destroy]
```

Comment:
Looks like Devise authentication filter is missing 
```suggestion
  before_action :authenticate_user!
  before_action :set_warehouse, only: [:show, :edit, :update, :destroy]
```